packages/web3-utils/test/fixtures/validation.ts

```diff
@@ -28,41 +28,43 @@ export const compareBlockNumbersValidData: [[Numbers, Numbers], number][] = [
 	[[1, BigInt(1)], 0],
 	[[1, BigInt(2)], -1],
 	[[2, BigInt(1)], 1],
-
 	[['earliest', 0], 0],
+	[[0, 'earliest'], 0],
+	[['earliest', 'earliest'], 0],
 	[['pending', 'pending'], 0],
 	[['latest', 'latest'], 0],
-	[['earliest', 2], -1],
 	[['earliest', 'pending'], -1],
 	[[BigInt('9007199254740992'), BigInt('9007199254740991')], 1],
 	[[13532346, 13532300], 1],
 	[['pending', 'latest'], 1],
-	[['latest', 0], 1],
-	[['latest', BigInt(1)], 1],
-	[['pending', 0], 1],
-	[['pending', BigInt(1)], 1],
 	[['safe', 'safe'], 0],
 	[['earliest', 'safe'], -1],
-	[['safe', 0], 1],
-	[[0, 'safe'], -1],
 	[['safe', 'pending'], -1],
 	[['pending', 'safe'], 1],
 	[['finalized', 'finalized'], 0],
 	[['earliest', 'finalized'], -1],
-	[['finalized', 0], 1],
 	[['finalized', 'pending'], -1],
-	[[0, 'finalized'], -1],
 	[['pending', 'finalized'], 1],
 	[['safe', 'latest'], -1],
 	[['latest', 'safe'], 1],
 ];
 
+const errorObj = new InvalidBlockError('Cannot compare blocktag with provided non-blocktag input.');
 export const compareBlockNumbersInvalidData: [[Numbers, Numbers], InvalidBlockError][] = [
-	[['pending', 'unknown'], new InvalidBlockError('unknown')],
-	[['', 'pending'], new InvalidBlockError('')],
-	[[22, 'finalized'], new InvalidBlockError('Cannot compare finalized tag with 22')],
-	[['finalized', 22], new InvalidBlockError('Cannot compare finalized tag with 22')],
-	[['genesis', 'finalized'], new InvalidBlockError('Genesis tag not supported')],
+	[['pending', 'unknown'], errorObj],
+	[['', 'pending'], errorObj],
+	[[22, 'finalized'], errorObj],
+	[['finalized', 22], errorObj],
+	[['earliest', 2], errorObj],
+	[[2, 'earliest'], errorObj],
+	[['latest', 110], errorObj],
+	[[222, 'latest'], errorObj],
+	[['pending', 230], errorObj],
+	[[10000, 'pending'], errorObj],
+	[['latest', BigInt(1)], errorObj],
+	[['pending', BigInt(1)], errorObj],
+	[['safe', 0], errorObj],
+	[[0, 'safe'], errorObj],
```

Comment:
```suggestion
```

Reply:
shared reason above, in some slim edge case safe can be 0.